### PR TITLE
Support for Modular RCS

### PR DESCRIPTION
**Pull request by al2me6**
Notably, `RCSDATA` has two additional keys `thrustTransformPositionOffset` (Vector3) and `thrustTransformScaleOffset` (float). These are necessary for proper (Waterfall) plume support.

Consider a modular RCS part with two models A and B, with B having a nozzle twice as big as A at the same scale. Thus, B needs to be at half scale to match A. On the other hand, the plume isn't aware of this scaling. Thus, since B is at half-scale, so is its thrust transform, and the plume, which is the right size for a, comes out as half the correct size on B. In this case, a `thrustTransformScaleOffset` of `2` applied to B will undo the outer rescaling and produce the right plume size.

`thrustTransformPositionOffset` is useful for fixing models with incorrectly positioned thrust transforms.

### CODE REVIEW

File: Source/ROLib/ModelDef/ROLModelModule.cs [hunk 1]
```diff
@@ -389,7 +389,14 @@ public void setupModel(bool doNotRescaleX = false)
...
         /// <param name="destinationName"></param>
-        public void RenameRCSThrustTransforms(string destinationName) => definition?.rcsModuleData?.RenameTransforms(root, destinationName);
+        public void RenameRCSThrustTransforms(string destinationName)
+        {
+            if (definition.rcsModuleData == null)
+            {
+                return;
+            }
+            definition.rcsModuleData.RenameTransforms(root, destinationName);
+        }
```
Comment (DRVeyl) [inline Source/ROLib/ModelDef/ROLModelModule.cs]: Why this change?  These are not Unity-managed classes and null-coalescing is valid here.

File: Source/ROLib/ModelDef/ROLModelModule.cs [hunk 2]
```diff
@@ -427,25 +434,17 @@ public void UpdateEngineModuleThrust(ModuleEngines engine, float thrustScalePowe
...
                 rcs.enableY = data.enableY;
                 rcs.enableZ = data.enableZ;
                 rcs.enablePitch = data.enablePitch;
                 rcs.enableYaw = data.enableYaw;
                 rcs.enableRoll = data.enableRoll;
             }
-            rcs.thrusterPower = power;
-            rcs.moduleIsEnabled = power > 0;
             rcs.OnStart(PartModule.StartState.Flying);
         }
```
Comment (DRVeyl) [inline Source/ROLib/ModelDef/ROLModelModule.cs]: Can you guarantee this is only ever called after the normal rcs OnStart?  And not at potentially inappropriate times such as before it has Loaded?

File: Source/ROLib/ModelDef/ROLModelDefinition.cs [hunk 3]
```diff
@@ -1,1096 +1,1115 @@
-﻿using System;
```
Comment (DRVeyl) [inline Source/ROLib/ModelDef/ROLModelDefinition.cs]: Why is this *entire file* diff'd?  Not sure how to review this.

**Review by DRVeyl — CHANGES_REQUESTED** [review]

**Comment by DRVeyl** [comment]
I guess I can go through the individual commits, though the conflicts in the branch need to be resolved.

File: Source/ROLib/ModelDef/ROLModelDefinition.cs (hunk 3 above)
Comment (siimav) [reply Source/ROLib/ModelDef/ROLModelDefinition.cs]: I imagine inconsistent line endings were normalized

File: Source/ROLib/ModelDef/ROLModelModule.cs (hunk 2 above)
Comment (DRVeyl) [reply Source/ROLib/ModelDef/ROLModelModule.cs]: Actually, I believe this doesn't have any place in the model manipulation.  It may have been relevant for setting the power values.  But that's now properly handled elsewhere, and so should whatever fixup is necessary to the running state be.

Comment (al2me6) [reply Source/ROLib/ModelDef/ROLModelModule.cs]: This I believe was inherited from whatever was in SSTU. But yes; I agree that it's quite fishy and should be moved. In fact there's code in ROE working _around_ this call.

Comment (DRVeyl) [reply Source/ROLib/ModelDef/ROLModelModule.cs]: I'll remove the entire method here.  Setting those enabled fields should go into the ROE code to update the module using the data in the definition.

**Comment by DRVeyl** [comment]
Ok.  Rebased on master, cleaned up some things, removed the commits of the dll [those can be done on master branch when it is time for release or as others prefer, but avoid doing them in feature branches and committing at least because of the conflicts it creates.]

@al2me6 Does this look sane?
